django/db/models/sql/compiler.py

```diff
@@ -146,9 +143,21 @@ def collapse_group_by(self, expressions, having):
                         getattr(expr, 'alias', None) == self.query.tables[0]):
                     pk = expr
                     break
+            # The logic here is: if the main model's primary key is in the
```

Comment:
Perhaps the comment could be less wordy, something like: `# If the main model's primary key is in the query, set new_expressions to that field and group by HAVING expressions and expressions...`. By the way, do you understand the meaning of "new_expressions" -- I don't see any thing in Django named that except this comment (seems to date to the original commit, 0c7633178fa9410f102e4708cef979b873bccb76).